### PR TITLE
[baremetal] Fix basic-build-test.sh to work in different env

### PR DESCRIPTION
**Pull request by jarvte**
fixed to work in xxx-arm-firmware repository

### CODE REVIEW

**Comment by jarvte** [comment]
When importing this fix to xxx-arm-firmware, also file scripts/output_env.sh must be imported

File: tests/scripts/basic-build-test.sh [hunk 1]
```diff
@@ -25,13 +25,41 @@
...
+        IS_UNIT_TEST=1
+    fi
+fi
+echo
+
+# Check test environment. If importer, src and inc folders are present then
+# there is a different directory structure and we need to adapt to it.
+if [ -d library -a -d include -a -d tests ] || [ -d importer -a -d inc -a -d src ]
+then
+    if [ -d importer -a -d inc -a -d src ]
```
Comment (mpg) [inline tests/scripts/basic-build-test.sh]: I find the logic a bit unclear with the nested if repeated part of the condition of the top-level if. I think the following structure might be clearer:
```
if [ -d importer -a -d inc -a -d src ]; then
    # adapt config
elif [ -d library -a -d include -a -d tests ]; then :; else
    # complain
fi

File: tests/scripts/basic-build-test.sh [hunk 2]
```diff
@@ -25,13 +25,41 @@
...
 #
 
 # Abort on errors (and uninitiliased variables)
 set -eu
 
-if [ -d library -a -d include -a -d tests ]; then :; else
+CONFIG_H='include/mbedtls/config.h'
+CONFIG_BAK="$CONFIG_H.bak"
+
+IS_UNIT_TEST=0
```
Comment (mpg) [inline tests/scripts/basic-build-test.sh]: I'm not sure I like the variable name here, I don't see what the "is" refers to. How about `UNIT_TESTS_ONLY` instead?

Comment (jarvte) [reply tests/scripts/basic-build-test.sh]: Changed as suggested

File: tests/scripts/basic-build-test.sh (hunk 1 above)
Comment (jarvte) [reply tests/scripts/basic-build-test.sh]: True, changed as suggested

**Comment by jarlamsa** [comment]
2 approving reviews, passing CI, ready for merge

**Pull request merged**